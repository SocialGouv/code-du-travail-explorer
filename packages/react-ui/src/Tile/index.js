import React from "react";
import PropTypes from "prop-types";
import { Button } from "../Button";
import { animations, box, fonts, spacings } from "../theme";
import styled from "styled-components";

const StyledTile = styled.a`
  position: relative;
  display: inline-flex;
  flex: 1 1 100%;
  flex-direction: column;
  flex-wrap: wrap;
  align-items: flex-start;
  justify-content: flex-end;
  box-sizing: border-box;
  margin: 0;
  padding: ${({ size }) =>
    size === "small"
      ? `${spacings.base} ${spacings.medium}`
      : `${spacings.medium} ${spacings.xmedium}`};
  color: ${({ theme }) => theme.paragraph};
  font-weight: 600;
  font-size: ${({ size }) =>
    size === "small" ? fonts.default : fonts.sizes.headings.small};
  text-align: left;
  text-decoration: none;
  background-color: ${({ theme, variant }) =>
    variant === "dark" ? theme.bgSecondary : theme.white};
  border: ${box.border};
  border-radius: ${box.borderRadius};
  cursor: pointer;
  transition: ${animations.transitionTiming} all;
  appearance: none;
  &:hover {
    color: ${({ theme }) => theme.paragraph};
    border-color: ${({ theme }) => theme.paragraph};
    box-shadow: ${box.shadow.default};
  }
`;

const OverflowWrapper = styled.div`
  display: block;
  max-width: 100%; /* ie11 fix :'( */
  padding-right: ${({ paddingRight }) => (paddingRight ? spacings.base : 0)};
  overflow-y: auto;
`;

const IconWrapper = styled.div`
  position: absolute;
  top: ${spacings.tiny};
  right: ${spacings.tiny};
`;

const ButtonWrapper = styled.div`
  margin-top: ${({ size }) =>
    size === "small" ? spacings.base : spacings.medium};
`;

export const Tile = React.forwardRef(
  ({ button, children, icon: Icon, size, ...props }, ref) => (
    <StyledTile
      as={props.href ? "a" : "button"}
      size={size}
      ref={ref}
      {...props}
    >
      {Icon && (
        <IconWrapper>
          <Icon />
        </IconWrapper>
      )}
      <OverflowWrapper paddingRight={Boolean(Icon)}>{children}</OverflowWrapper>
      {button && (
        <ButtonWrapper size={size}>
          <Button as="div" small={size === "small"} variant="secondary">
            {button}
          </Button>
        </ButtonWrapper>
      )}
    </StyledTile>
  )
);

Tile.displayName = "Tile";

Tile.propTypes = {
  button: PropTypes.string,
  children: PropTypes.node.isRequired,
  href: PropTypes.string,
  icon: PropTypes.elementType,
  size: PropTypes.string,
  variant: PropTypes.string
};

Tile.defaultProps = {
  href: "",
  button: "",
  size: "",
  variant: ""
};
