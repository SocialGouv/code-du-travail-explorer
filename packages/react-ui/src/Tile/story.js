import React from "react";

import { Section } from "../layout/Section";
import { Holidays, Time, Salary } from "../icons";
import { Tile } from ".";

export default {
  component: Tile,
  title: "Components|Tile"
};

export const base = () => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      alignItems: "stretch",
      width: "270px",
      maxWidth: "100%"
    }}
  >
    <Section>
      <Tile href="#" title="Basic tile" />
    </Section>
    <Section>
      <Tile href="#" icon={Time} title="Basic tile">
        With some basic content only to see how it behaves
      </Tile>
    </Section>
    <Section>
      <Tile href="#" custom title="Custom tile with purposely long title">
        Make sure there is no collision with custom icon. Beware of mobile.
      </Tile>
    </Section>
    <Section>
      <Tile href="#" icon={Salary} title="Tile with an icon" />
    </Section>
    <Section>
      <Tile href="#" icon={Holidays} custom title="Tile custom with an icon">
        And some basic content only to see how it behaves
      </Tile>
    </Section>
    <Section>
      <Tile title="Button tile, no href prop, should have an onClick" />
    </Section>
  </div>
);

export const wide = () => (
  <div
    style={{
      display: "flex",
      flexDirection: "column",
      alignItems: "stretch",
      width: "600px",
      maxWidth: "100%"
    }}
  >
    <Section>
      <Tile wide href="#" title="Wide tile" subtitle="Some theme here " />
    </Section>
    <Section>
      <Tile wide href="#" title="Wide tile">
        With some basic content only to see how it behaves
      </Tile>
    </Section>
    <Section>
      <Tile
        wide
        href="#"
        custom
        title="Custom tile with purposely long title to see how it goes with custom icon"
      >
        Make sure there is no collision with custom icon. Beware of mobile.
      </Tile>
    </Section>
    <Section>
      <Tile
        wide
        href="#"
        custom
        subtitle="Custom tile with purposely long title to see how it goes with custom icon"
        title="Some title"
      >
        Make sure there is no collision with custom icon. Beware of mobile.
      </Tile>
    </Section>
    <Section>
      <Tile wide title="Button tile, no href prop, should have an onClick" />
    </Section>
  </div>
);
