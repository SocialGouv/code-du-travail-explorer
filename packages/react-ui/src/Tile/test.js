import React from "react";
import { render } from "@testing-library/react";

import { Time } from "../icons";
import { Tile } from ".";

describe("<Tile />", () => {
  it("renders a tile with text inside", () => {
    const { container } = render(
      <Tile title="Hello !">There is text inside</Tile>
    );
    expect(container).toMatchSnapshot();
    expect(container.getElementsByTagName("button").length).toBe(1);
  });
  it("renders a tile with a custom badge", () => {
    const { container } = render(
      <Tile title="Hello !">There is text inside</Tile>
    );
    expect(container).toMatchSnapshot();
    expect(container.getElementsByTagName("button").length).toBe(1);
  });
  it("renders a tile with an icon", () => {
    const { container } = render(
      <Tile title="Hello !" icon={Time}>
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
    expect(container.getElementsByTagName("button").length).toBe(1);
  });
  it("renders a tile with a stripe", () => {
    const { container } = render(
      <Tile title="Hello !" striped>
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
  });
  it("renders a wide tile with a subtitle", () => {
    const { container } = render(
      <Tile subtitle="This is a theme" title="Hello !">
        There is text inside
      </Tile>
    );
    expect(container).toMatchSnapshot();
    expect(container.getElementsByTagName("button").length).toBe(1);
  });
});
