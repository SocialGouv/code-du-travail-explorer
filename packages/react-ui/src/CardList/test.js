import React from "react";
import { render } from "@testing-library/react";
import { CardList } from ".";
import { Tile } from "../Tile";

const items = Array.from({ length: 7 }, (_, index) => ({
  title: `Title ${index + 1}`,
  id: `${index + 1}`
}));

describe("<CardList />", () => {
  test("should render a collection with a title and description", () => {
    const { container } = render(
      <CardList title="Hello" desc="Hello Tiles">
        {items.map(item => (
          <Tile key={item.id}>{item.title}</Tile>
        ))}
      </CardList>
    );
    expect(container).toMatchSnapshot();
  });
  test("should render a signle title", () => {
    const { container } = render(
      <CardList title="Hello">
        <Tile href="#">coucou</Tile>
      </CardList>
    );
    expect(container).toMatchSnapshot();
  });
  test("should render a collection with a title", () => {
    const { container } = render(
      <CardList title="Hello">
        {items.map(item => (
          <Tile key={item.id}>{item.title}</Tile>
        ))}
      </CardList>
    );
    expect(container).toMatchSnapshot();
  });
  test("should render a collection with a title with a link and a desc", () => {
    const { container } = render(
      <CardList
        title="CDTN"
        href="https://code.travail.gouv.fr"
        desc="le code du travail numÃ©rique"
      >
        {items.map(item => (
          <Tile key={item.id} title={item.title} />
        ))}
      </CardList>
    );
    expect(container).toMatchSnapshot();
  });
  test("should render a collection on 3column", () => {
    const { container } = render(
      <CardList title="Hello" columns={3}>
        {items.map(item => (
          <Tile key={item.id} title={item.title} />
        ))}
      </CardList>
    );
    expect(container).toMatchSnapshot();
  });
});
