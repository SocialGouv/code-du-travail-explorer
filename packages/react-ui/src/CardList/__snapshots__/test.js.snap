// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardList /> should render a collection on 3column 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 1.25rem;
  margin-right: calc(-1 * 0.625rem);
  margin-bottom: 2rem;
  margin-left: calc(-1 * 0.625rem);
  padding: 0;
  list-style-type: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: calc(100% / 3 - 2 * 0.625rem);
  margin: 0.625rem;
}

.c0 {
  margin-bottom: 0;
  color: #005994;
  font-weight: 600;
  font-size: 1.375rem;
  font-family: Muli,-apple-system,BlinkMacSystemFont,"Helvetica Neue", Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c3 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  box-sizing: border-box;
  margin: 0;
  padding: 1.25rem 1.5rem;
  color: #005994;
  font-weight: bold;
  font-size: 1.125rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: 250ms all;
  transition: 250ms all;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c3:hover {
  border-color: #005994;
  box-shadow: 0 5px 10px 0 #ebeff3;
}

.c4 {
  display: block;
  max-width: 100%;
  padding-right: 0;
  overflow-y: auto;
}

@media (max-width:980px) {
  .c2 {
    width: calc( 100% / 2 - 2 * 0.625rem );
  }
}

@media (max-width:600px) {
  .c2 {
    width: calc( 100% - 1 * 0.625rem );
  }
}

<div>
  <div>
    <h2
      class="c0"
    >
      Hello
    </h2>
    
  </div>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 1
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 2
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 3
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 4
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 5
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 6
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 7
        </div>
        
      </button>
    </li>
  </ul>
</div>
`;

exports[`<CardList /> should render a collection with a title 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 1.25rem;
  margin-right: calc(-1 * 0.625rem);
  margin-bottom: 2rem;
  margin-left: calc(-1 * 0.625rem);
  padding: 0;
  list-style-type: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: calc(100% / 4 - 2 * 0.625rem);
  margin: 0.625rem;
}

.c0 {
  margin-bottom: 0;
  color: #005994;
  font-weight: 600;
  font-size: 1.375rem;
  font-family: Muli,-apple-system,BlinkMacSystemFont,"Helvetica Neue", Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c3 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  box-sizing: border-box;
  margin: 0;
  padding: 1.25rem 1.5rem;
  color: #005994;
  font-weight: bold;
  font-size: 1.125rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: 250ms all;
  transition: 250ms all;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c3:hover {
  border-color: #005994;
  box-shadow: 0 5px 10px 0 #ebeff3;
}

.c4 {
  display: block;
  max-width: 100%;
  padding-right: 0;
  overflow-y: auto;
}

@media (max-width:980px) {
  .c2 {
    width: calc( 100% / 3 - 2 * 0.625rem );
  }
}

@media (max-width:600px) {
  .c2 {
    width: calc( 100% - 2 * 0.625rem );
  }
}

<div>
  <div>
    <h2
      class="c0"
    >
      Hello
    </h2>
    
  </div>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 1
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 2
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 3
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 4
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 5
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 6
        </div>
        
      </button>
    </li>
    <li
      class="c2"
    >
      <button
        class="c3"
        href=""
      >
        <div
          class="c4"
        >
          Title 7
        </div>
        
      </button>
    </li>
  </ul>
</div>
`;

exports[`<CardList /> should render a collection with a title and description 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 1.25rem;
  margin-right: calc(-1 * 0.625rem);
  margin-bottom: 2rem;
  margin-left: calc(-1 * 0.625rem);
  padding: 0;
  list-style-type: none;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: calc(100% / 4 - 2 * 0.625rem);
  margin: 0.625rem;
}

.c0 {
  margin-bottom: 0;
  color: #005994;
  font-weight: 600;
  font-size: 1.375rem;
  font-family: Muli,-apple-system,BlinkMacSystemFont,"Helvetica Neue", Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c1 {
  color: #005994;
  font-weight: 300;
  font-size: 1rem;
  font-family: Muli,-apple-system,BlinkMacSystemFont,"Helvetica Neue", Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c4 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  box-sizing: border-box;
  margin: 0;
  padding: 1.25rem 1.5rem;
  color: #005994;
  font-weight: bold;
  font-size: 1.125rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: 250ms all;
  transition: 250ms all;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4:hover {
  border-color: #005994;
  box-shadow: 0 5px 10px 0 #ebeff3;
}

.c5 {
  display: block;
  max-width: 100%;
  padding-right: 0;
  overflow-y: auto;
}

@media (max-width:980px) {
  .c3 {
    width: calc( 100% / 3 - 2 * 0.625rem );
  }
}

@media (max-width:600px) {
  .c3 {
    width: calc( 100% - 2 * 0.625rem );
  }
}

<div>
  <div>
    <h2
      class="c0"
    >
      Hello
    </h2>
    <div
      class="c1"
    >
      Hello Tiles
    </div>
  </div>
  <ul
    class="c2"
  >
    <li
      class="c3"
    >
      <button
        class="c4"
        href=""
      >
        <div
          class="c5"
        >
          Title 1
        </div>
        
      </button>
    </li>
    <li
      class="c3"
    >
      <button
        class="c4"
        href=""
      >
        <div
          class="c5"
        >
          Title 2
        </div>
        
      </button>
    </li>
    <li
      class="c3"
    >
      <button
        class="c4"
        href=""
      >
        <div
          class="c5"
        >
          Title 3
        </div>
        
      </button>
    </li>
    <li
      class="c3"
    >
      <button
        class="c4"
        href=""
      >
        <div
          class="c5"
        >
          Title 4
        </div>
        
      </button>
    </li>
    <li
      class="c3"
    >
      <button
        class="c4"
        href=""
      >
        <div
          class="c5"
        >
          Title 5
        </div>
        
      </button>
    </li>
    <li
      class="c3"
    >
      <button
        class="c4"
        href=""
      >
        <div
          class="c5"
        >
          Title 6
        </div>
        
      </button>
    </li>
    <li
      class="c3"
    >
      <button
        class="c4"
        href=""
      >
        <div
          class="c5"
        >
          Title 7
        </div>
        
      </button>
    </li>
  </ul>
</div>
`;

exports[`<CardList /> should render a collection with a title with a link and a desc 1`] = `
.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 1.25rem;
  margin-right: calc(-1 * 0.625rem);
  margin-bottom: 2rem;
  margin-left: calc(-1 * 0.625rem);
  padding: 0;
  list-style-type: none;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: calc(100% / 4 - 2 * 0.625rem);
  margin: 0.625rem;
}

.c0 {
  margin-bottom: 0;
  color: #005994;
  font-weight: 600;
  font-size: 1.375rem;
  font-family: Muli,-apple-system,BlinkMacSystemFont,"Helvetica Neue", Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c4 {
  color: #005994;
  font-weight: 300;
  font-size: 1rem;
  font-family: Muli,-apple-system,BlinkMacSystemFont,"Helvetica Neue", Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #005994;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2:hover {
  color: #006be6;
}

.c3 {
  margin-left: 1rem;
  color: #006be6;
  -webkit-transition: 250ms ease-out -webkit-transform;
  -webkit-transition: 250ms ease-out transform;
  transition: 250ms ease-out transform;
}

.c1:hover .c3 {
  -webkit-transform: translateX(7px);
  -ms-transform: translateX(7px);
  transform: translateX(7px);
}

.c7 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  box-sizing: border-box;
  margin: 0;
  padding: 1.25rem 1.5rem;
  color: #005994;
  font-weight: bold;
  font-size: 1.125rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: 250ms all;
  transition: 250ms all;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c7:hover {
  border-color: #005994;
  box-shadow: 0 5px 10px 0 #ebeff3;
}

.c8 {
  display: block;
  max-width: 100%;
  padding-right: 0;
  overflow-y: auto;
}

@media (max-width:980px) {
  .c6 {
    width: calc( 100% / 3 - 2 * 0.625rem );
  }
}

@media (max-width:600px) {
  .c6 {
    width: calc( 100% - 2 * 0.625rem );
  }
}

<div>
  <div>
    <h2
      class="c0"
    >
      <a
        class="c1 c2"
        href="https://code.travail.gouv.fr"
        title="le code du travail numérique"
      >
        CDTN
        <svg
          aria-hidden="true"
          class="c3"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <polyline
            points="9 18 15 12 9 6"
          />
        </svg>
      </a>
    </h2>
    <div
      class="c4"
    >
      le code du travail numérique
    </div>
  </div>
  <ul
    class="c5"
  >
    <li
      class="c6"
    >
      <button
        class="c7"
        href=""
      >
        <div
          class="c8"
        >
          Title 1
        </div>
        
      </button>
    </li>
    <li
      class="c6"
    >
      <button
        class="c7"
        href=""
      >
        <div
          class="c8"
        >
          Title 2
        </div>
        
      </button>
    </li>
    <li
      class="c6"
    >
      <button
        class="c7"
        href=""
      >
        <div
          class="c8"
        >
          Title 3
        </div>
        
      </button>
    </li>
    <li
      class="c6"
    >
      <button
        class="c7"
        href=""
      >
        <div
          class="c8"
        >
          Title 4
        </div>
        
      </button>
    </li>
    <li
      class="c6"
    >
      <button
        class="c7"
        href=""
      >
        <div
          class="c8"
        >
          Title 5
        </div>
        
      </button>
    </li>
    <li
      class="c6"
    >
      <button
        class="c7"
        href=""
      >
        <div
          class="c8"
        >
          Title 6
        </div>
        
      </button>
    </li>
    <li
      class="c6"
    >
      <button
        class="c7"
        href=""
      >
        <div
          class="c8"
        >
          Title 7
        </div>
        
      </button>
    </li>
  </ul>
</div>
`;

exports[`<CardList /> should render a signle title 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  margin-top: 1.25rem;
  margin-right: calc(-1 * 0.625rem);
  margin-bottom: 2rem;
  margin-left: calc(-1 * 0.625rem);
  padding: 0;
  list-style-type: none;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: calc(100% / 4 - 2 * 0.625rem);
  margin: 0.625rem;
}

.c0 {
  margin-bottom: 0;
  color: #005994;
  font-weight: 600;
  font-size: 1.375rem;
  font-family: Muli,-apple-system,BlinkMacSystemFont,"Helvetica Neue", Helvetica,Arial,sans-serif;
  line-height: 1.4;
}

.c3 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-flex: 1 1 100%;
  -ms-flex: 1 1 100%;
  flex: 1 1 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  box-sizing: border-box;
  margin: 0;
  padding: 1.25rem 1.5rem;
  color: #005994;
  font-weight: bold;
  font-size: 1.125rem;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  background-color: #fff;
  border: 1px solid #c9d3df;
  border-radius: 0.25rem;
  cursor: pointer;
  -webkit-transition: 250ms all;
  transition: 250ms all;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c3:hover {
  border-color: #005994;
  box-shadow: 0 5px 10px 0 #ebeff3;
}

.c4 {
  display: block;
  max-width: 100%;
  padding-right: 0;
  overflow-y: auto;
}

@media (max-width:980px) {
  .c2 {
    width: calc( 100% / 3 - 2 * 0.625rem );
  }
}

@media (max-width:600px) {
  .c2 {
    width: calc( 100% - 2 * 0.625rem );
  }
}

<div>
  <div>
    <h2
      class="c0"
    >
      Hello
    </h2>
    
  </div>
  <ul
    class="c1"
  >
    <li
      class="c2"
    >
      <a
        class="c3"
        href="#"
      >
        <div
          class="c4"
        >
          coucou
        </div>
        
      </a>
    </li>
  </ul>
</div>
`;
